byoqm/source_repository/source_repository.py

```diff
@@ -14,55 +24,103 @@ class SourceRepository:
     under analysis
     """
 
-    def __init__(self, src_root: Path, language: str):
-        if src_root.is_file():
-            self.src_paths = [src_root]
-        else:
-            self.src_paths = [file for file in src_root.glob(languages[language])]
-
-        self.tree_sitter_language = Language(_TREESITTER_BUILD.__str__(), language)
-        logging.info(f"Language set to: {language}")
-        self.asts = {}
-        self.language = language
-        self._parser = Parser()
-        self._parser.set_language(self.tree_sitter_language)
-        self.file_encodings = self._get_encodings(self.src_paths)
-
-    def getAst(self, for_file: Path) -> tree_sitter.Tree:
+    def __init__(self, src_root: Path):
+        self.src_root: Path = src_root
+        self.asts: Dict[Path, tree_sitter.Tree] = {}
+        self.files: Dict[Path, FileInfo] = self._discover_files()
+        self.tree_sitter_languages: Dict[str, tree_sitter.Language] = {
+            "python": tree_sitter.Language(_TREESITTER_BUILD, "python"),
+            "c_sharp": tree_sitter.Language(_TREESITTER_BUILD, "c_sharp"),
+            "java": tree_sitter.Language(_TREESITTER_BUILD, "java"),
+        }
+
+        python_parser = tree_sitter.Parser()
+        python_parser.set_language(self.tree_sitter_languages["python"])
+
+        c_sharp_parser = tree_sitter.Parser()
+        c_sharp_parser.set_language(self.tree_sitter_languages["c_sharp"])
+
+        java_parser = tree_sitter.Parser()
+        java_parser.set_language(self.tree_sitter_languages["java"])
+
+        self.tree_sitter_parsers: Dict[str, tree_sitter.Parser] = {
+            "python": python_parser,
+            "c_sharp": c_sharp_parser,
+            "java": java_parser,
+        }
```

Comment:
We need to re-write the documentation of adding supported languages to the parser

Reply:
Shouldn't this be dealt with in #96, since the documentation isn't introduced on this PR yet.

Reply:
yeye, I think I wrote this as more of a mental note